File: factory/Visa.java
package factory;

public class Visa extends InternationalPaymentMethod {

    public Visa(String cardHolder, String cardNumber, String cvv, String expiryDate) {
        super(cardHolder, cardNumber, cvv, expiryDate);
    }

    @Override
    public void authorizePayment() {
        System.out.println("Authorizing visa card....");
    }

    @Override
    public void startMoneyTransfer() {
        System.out.println("Start money transferring for a visa card....");
    }

    @Override
    public void calculatePaymentFees(double amount) {
        System.out.println("Calculating visa card payment fees....");
    }
}

File: typescript/factory/Visa.ts
import { InternationalPaymentMethod } from "./international-payment-method";

export class Visa extends InternationalPaymentMethod {
    constructor(
      cardHolder: string,
      cardNumber: string,
      cvv: string,
      expiryDate: string
    ) {
      super(cardHolder, cardNumber, cvv, expiryDate);
    }
  
    authorizePayment(): void {
      console.log("Authorizing Visa card....");
    }
  
    startMoneyTransfer(): void {
      console.log("Start money transferring for a Visa card....");
    }
  
    calculatePaymentFees(amount: number): void {
      console.log("Calculating Visa card payment fees....");
    }
  }